Maple2.File.Ingest/Mapper/TableMapper.cs

```diff
@@ -1111,6 +1114,53 @@ public class TableMapper : TypeMapper<TableMetadata> {
         return new Color(color.B, color.G, color.R, color.A);
     }
 
+    private MeretMarketCategoryTable ParseMeretMarketCategoryTable() {
+        string feature = FeatureLocaleFilter.FeatureEnabled("MeratMarketNewBM") ? "MeratMarketNewBM" : string.Empty;
+        Dictionary<int, IReadOnlyDictionary<int, MeretMarketCategoryTable.Tab>> results = GetMereMarketTabs(feature);
+        return new MeretMarketCategoryTable(results);
+	}
+
+    private Dictionary<int, IReadOnlyDictionary<int, MeretMarketCategoryTable.Tab>> GetMereMarketTabs(string featureEnabled = "") {
+        var results = new Dictionary<int, IReadOnlyDictionary<int, MeretMarketCategoryTable.Tab>>();
+        foreach ((int id, string feature, MeretMarketCategory category) in parser.ParseMeretMarketCategory()) {
```

Comment:
idk why feature is being returned here, but you **never** need to filter by feature manually. This is already done by the parsing framework.

There's a reason the variable name begins with `_`, it means don't use it.
`yield return (category.id, environment._feature, category);`

The problem seems to be `XmlElement` instead of `M2dFeatureLocale`: https://github.com/kOchirasu/Maple2.File/blob/443583860cdb2d9f5e85dd1d49facac208cb5d7e/Maple2.File.Parser/Xml/Table/MeretMarketCategory.cs#L11

Reply:
It's because the xml is written a bit different:
`	<environment feature="MeratMarketNewBM" locale="">
		<category id="100000" tabType="sort">
			<tab name="40300" id="40300" symbol="UIMeratMarketDialog_symbol_etc" priority="6" static="1" />
			<tab name="40600" id="40600" symbol="UIMeratMarketDialog_symbol_instrument" priority="9" static="1" />
		</category>
	</environment>`

Reply:
I don't understand what's different

Reply:
It's wrapped in environment, while the "normal" one isn't.
for comparison:
https://github.com/Zintixx/MapleStory2-English/blob/master/Latest-GMS2/table/na/meratmarketcategory.xml

Reply:
It would, but I would still need to do a check and ingest the correct section if the feature is enabled, wouldn't I?

Reply:
You should not because `environment` would be empty if the feature is not enabled.

Reply:
**If the feature wasn't enabled:**

It would not ingest environment and only parse the "normal" meret market.

**If the feature was enabled:**
It would ingest both. You would still need some kind of check whether it should ignore the "normal" meret market.

I understand this is how it should be properly parsed but it doesn't eliminate the need to check feature enabled when ingesting.

Reply:
We can also add the check in the parser itself:
```cs
if (environment.Count > 0) {
    // return environment results
}
// return normal results
```

Reply:
Okay sure we can do that. Did you want to include this on your PR for Maple2.File or I could open one?

Reply:
I'll make the change since I have it already

Reply:
resolved